### PR TITLE
In memory blaze problem view filtering

### PR DESCRIPTION
**Pull request by sha1n**
This PR makes the Blaze Problems view "Show errors only" toggle button work on an existing view, and no longer requires a sync in order to take effect. 

I couldn't figure out an easy way to filter the existing views, without a major refactoring, which involves changes that doesn't sit well with the existing design. So I tried to fit into the existing design as smoothly as possible, while keeping the existing entities' responsibilities and dependencies.

**Main changes:**
1. Introduced `BlazeProblemsSink`, which is an in-memory container, to which all reported problems are added.
2. Changed the existing `ShowOnlyErrorsAction` which handles the toggle button, to trigger a reload on `BlazeProblemsView`
3. `BlazeProblemsView.reload()` clears the `BlazeProblemsViewPanel` tree structure that hosts the data and repopulates it from the data held by `BlazeProblemsSink`

@hmemcpy please review and let me know what you think.
@ittaiz - FYI, in case you wanna take a look

### CODE REVIEW

File: base/src/com/google/idea/blaze/base/ui/problems/BlazeProblemsView.java [hunk 1]
```diff
@@ -46,16 +46,9 @@
 import org.jetbrains.annotations.NotNull;
 
 import javax.annotation.Nullable;
-import javax.swing.Icon;
+import javax.swing.*;
```
Comment (ittaiz) [inline base/src/com/google/idea/blaze/base/ui/problems/BlazeProblemsView.java]: drive-by: google style guide disallows wildcard imports

Comment (sha1n) [reply base/src/com/google/idea/blaze/base/ui/problems/BlazeProblemsView.java]: Thanks! I'll fix that.
@hmemcpy do you know whether there is an intellij code-style settings I can use to make sure I don't violate it again? Because the next time I optimize imports, it will happen again..

**Comment by hmemcpy** [comment]
The `BlazeProblemsSink` is the exact thing I need to @idankWix's feature as well - with one addition - there should be an event raised with the issue on the messageBus.
But I'm hopeful this change makes it in, because we were about to propose a similar one!

**Comment by hmemcpy** [comment]
Other than Ittai's nits, it LGTM. But I do want to incorporate this event message to the sink:
https://github.com/wix-playground/intellij/blob/bfdd75830f7b669205346aef1e921bb1cb610909/base/src/com/google/idea/blaze/base/ui/problems/BlazeProblemsView.java#L188-L190

**Comment by sha1n** [comment]
Thanks @hmemcpy! I'm not sure my change will be accepted, so I wouldn't build anything on top of it at the moment. I'll PR the main repo and lets see what they have to say. Agree?

**Comment by hmemcpy** [comment]
Sure! Before that, did you base your change off `master` or `wix-master`? Because `wix-master` is currently broken due to Google removing a needed class. So if you could branch your change off `bazelbuild/master` instead, it would be much easier to PR.

**Comment by sha1n** [comment]
Closing since this feature has been implemented upstream.
